src/librustdoc/config.rs

```diff
@@ -320,7 +320,15 @@ pub(crate) enum EmitType {
     Unversioned,
     Toolchain,
     InvocationSpecific,
-    DepInfo(Option<PathBuf>),
+    DepInfo(Option<OutFile>),
+}
+
+/// Similar to [`rustc_session::config::OutFileName`] but can't use that
+/// because it doesn't implement `Eq`.
+#[derive(Clone, Debug, PartialEq, Eq)]
+pub(crate) enum OutFile {
```

Comment:
Feel free to make rustc's OutFile `Eq` (and pub if it isn't already) and use it here. We already reuse rustc's Input type.